### PR TITLE
feat(InlineLoading): add support for warning status

### PR DESCRIPTION
**Pull request by janhassel**
This PR adds the option for a warning status to the InlineLoading component.

#### Changelog

**New**

- `warning` status in InlineLoading

**Changed**

- Changed current error icon to ErrorFilled icon to be consistent with the filled `finished` state and improve contrast for warning.

### CODE REVIEW

**Comment by asudoh** [comment]
New UX feature, requires design review.

File: packages/react/src/components/InlineLoading/InlineLoading.js [hunk 1]
```diff
@@ -41,8 +48,7 @@ export default function InlineLoading({
           className={`${prefix}--inline-loading__checkmark-container`}
         />
       );
-    }
-    if (status === 'inactive' || status === 'active') {
+    } else if (status === 'inactive' || status === 'active') {
```
Comment (emyarod) [inline packages/react/src/components/InlineLoading/InlineLoading.js]: since the `if` block contains a return statement the `else` is unneeded

```suggestion
    } 
    if (status === 'inactive' || status === 'active') {
```

File: packages/react/src/components/InlineLoading/InlineLoading.js [hunk 2]
```diff
@@ -28,9 +32,12 @@ export default function InlineLoading({
...
     if (status === 'error') {
-      return <Error20 className={`${prefix}--inline-loading--error`} />;
-    }
-    if (status === 'finished') {
+      return <ErrorFilled16 className={`${prefix}--inline-loading--error`} />;
+    } else if (status === 'warning') {
+      return (
+        <WarningFilled16 className={`${prefix}--inline-loading--warning`} />
+      );
+    } else if (status === 'finished') {
```
Comment (emyarod) [inline packages/react/src/components/InlineLoading/InlineLoading.js]: since the `if` block contains a return statement the `else` is unneeded

```suggestion
    }
    if (status === 'finished') {
```

File: packages/react/src/components/InlineLoading/InlineLoading.js [hunk 3]
```diff
@@ -28,9 +32,12 @@ export default function InlineLoading({
   const loadingClasses = classNames(`${prefix}--inline-loading`, className);
   const getLoading = () => {
     if (status === 'error') {
-      return <Error20 className={`${prefix}--inline-loading--error`} />;
-    }
-    if (status === 'finished') {
+      return <ErrorFilled16 className={`${prefix}--inline-loading--error`} />;
+    } else if (status === 'warning') {
```
Comment (emyarod) [inline packages/react/src/components/InlineLoading/InlineLoading.js]: since the `if` block contains a return statement the `else` is unneeded

```suggestion
    }
    if (status === 'warning') {
```

**Comment by janhassel** [comment]
@emyarod Oh yeah, right. I was wondering why it wasn't else-if and didn't think about the returns. Thank you!

**Comment by joshblack** [comment]
Hey @janhassel! Is there a corresponding issue we could look at for this feature?

**Review by aagonzales — APPROVED** [review]
Agreed, should be the filled version.

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @janhassel!

**Comment by janhassel** [comment]
@joshblack We haven't opened an issue for this, but I can send you some information about the use case via Slack if you want.

**Comment by joshblack** [comment]
@janhassel would you mind opening a ticket for it just so we can better understand the use-case/goal for adding it in?

Separately, I think the changes to the error icons and what-not are perfect so it might make sense to get that in with this PR and then do the net-new functionality in a later one once we understand the use-case.

**Comment by janhassel** [comment]
@joshblack I opened an issue with a sample use case (committing in git): #4835

**Comment by joshblack** [comment]
Thanks @janhassel!

**Comment by joshblack** [comment]
Just a quick note, still waiting for another dev review and for the linked issue to be signed off by design 👍 After talking in our support channel it seems like @aagonzales might have only noticed the icon change (which is perfect 💯) and not the additional feature so will wait on them to verify in the linked issue!

**Comment by aagonzales** [comment]
@janhassel If the user is given the warning does that mean the load was not successful? The example is about merge conflicts but I'm trying to understand what is being recommended for the general guidance of when "warning" would be used instead of a binary success/fail.

**Comment by janhassel** [comment]
@aagonzales Let's think about another example: a file uploader. We show the user the Inline Loading component while the file is being uploaded, once it's done, we switch to the `finished` state. If the upload failed, we show the `error` state.
But if a file with this name already exists on the destination, we can show the user a warning state, telling them that their file has been uploaded but they need to tell us whether or not to overwrite the existing file.

So to sum it up:
**success / finished**
- everything worked, nothing to do

**error**
- something didn't work out, the user can not proceed with the aspired action

**warning**
- it worked, but due to an issue there is an extra step required for the user to accomplish their action
- something didn't work out, but it is not crucial and the user can proceed. The action might be accomplished later on by the system.

**Review by tw15egan — APPROVED** [review]
Looks great on the dev side of things, just need @aagonzales to sign off and we can merge it in 👍

**Comment by joshblack** [comment]
Any chance we could split this into just the icon change and then the proposal for the behavior? Would love to chat/learn more about the warning use-case and how we can incorporate into docs/guidance 👍

**Comment by janhassel** [comment]
Definitely, I opened another PR just concerning the icon: #5018

@joshblack I'll send you a detailed example on Slack.

**Comment by asudoh** [comment]
Any updates @janhassel @joshblack? Thanks!

**Comment by aagonzales** [comment]
Controversial statement: I think this feature is its own component and not an enhancement to inline loading. It's a status indictor and not solely about the state of loading. Jan explained the scenario to me further and I believe its a valid use-case I'm just not sure it belongs within the `inline loading` component. 

image

**Comment by asudoh** [comment]
@aagonzales Would you have a chance to other designer peers and see what we should do with this PR...? Thanks!

**Comment by joshblack** [comment]
Closing this out in favor of the proposal over in: https://github.com/carbon-design-system/carbon/issues/4835

Thanks again for talking through this stuff @janhassel!
